pyproject.toml

```diff
@@ -7,43 +7,133 @@ license = "MIT"
 readme = "README.rst"
 repository = "https://github.com/useblocks/sphinx-performance"
 documentation = "https://sphinx-performance.readthedocs.io/en/latest/"
-classifiers=[
-        'Development Status :: 5 - Production/Stable',
-        'Intended Audience :: Developers',
-        'License :: OSI Approved :: MIT License',
-        'Operating System :: OS Independent',
-        'Programming Language :: Python',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10',
-        'Programming Language :: Python :: 3.11',
-        'Topic :: Documentation',
-        'Topic :: Utilities',
-        'Framework :: Sphinx',
-    ]
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: MIT License',
+    'Operating System :: OS Independent',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Topic :: Documentation',
+    'Topic :: Utilities',
+    'Framework :: Sphinx',
+]
 
 [tool.poetry.dependencies]
-python = "^3.7.0"
+python = ">=3.8,<4"
 click = "^8.0.3"
-rich = "^11.2.0"
 Jinja2 = "^3.0.3"
 memray = "^1.3.1"
-sphinx = "^5.3.0"
 pyinstrument = "^4.3.0"
+rich = "^11.2.0"
 snakeviz = "^2.1.1"
+sphinx = "^5.3.0"
 
-[tool.poetry.dev-dependencies]
+[tool.poetry.group.dev.dependencies]
+black = "^23.7.0"
+ruff = "^0.0.285"
+
+[tool.poetry.group.docs.dependencies]
+# some shared dependencies as the app itself is called in the docs build
+click = "^8.0.3"
+memray = "^1.3.1"
+pyinstrument = "^4.3.0"
+rich = "^11.2.0"
+sphinx = "^5.3.0"
+sphinxcontrib-programoutput = "^0.17"
 
 [tool.poetry.scripts]
-sphinx-performance = 'sphinx_performance.performance:cli_performance'
 sphinx-analysis = 'sphinx_performance.analysis:cli_analysis'
+sphinx-performance = 'sphinx_performance.performance:cli_performance'
+
+[tool.ruff]
+select = ["ALL"] # Enable all checks and maintain an ignore list
+# a lot of ignores here due to activation of ruff linting in an established project
+# candidates for reactivation are marked with a star
+ignore = [
```

Comment:
I guess some of these can be removed later?

Reply:
Yes, all the ones with a star as mentioned in my comment.